src/vs/workbench/contrib/tasks/common/jsonSchema_v2.ts

```diff
@@ -94,14 +94,19 @@ const detail: IJSONSchema = {
 	description: nls.localize('JsonSchema.tasks.detail', 'An optional description of a task that shows in the Run Task quick pick as a detail.')
 };
 
-const color: IJSONSchema = {
-	type: 'string',
-	description: nls.localize('JsonSchema.tasks.color', 'An optional color for the task icon')
-};
-
 const icon: IJSONSchema = {
-	type: 'string',
-	description: nls.localize('JsonSchema.tasks.icon', 'An optional icon for the task')
+	type: 'object',
+	description: nls.localize('JsonSchema.tasks.icon', 'An optional icon for the task'),
+	properties: {
+		id: {
+			type: 'string',
+			description: nls.localize('JsonSchema.tasks.icon.id', 'The codicon id')
```

Comment:
Is the expected format `"zap"` or `"$(zap)"`? For reference, in package.json in places we support codicons it's `"$(zap)"`.

Reply:
I think `$()` is the preferred format but if I remember right the terminal API supports both

Reply:
Terminal API doesn't support $() but can trim that off before it's added to the shell launch config if we think that's better

Reply:
This is what I meant by API using $():

https://github.com/microsoft/vscode-js-debug/blob/f314b55c8035fa535c42411c522714124df2dce5/package.json#L195

Reply:
we must trim that somewhere bc we wrap the icon like this 

https://github.com/microsoft/vscode/blob/b278fdaf7b3f2f3ac8df841b1b232b83e706cca2/src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts#L305